src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -341,11 +353,41 @@
     (->> (map format-task info-from-jruby)
          (middleware-utils/json-response 200))))
 
+(schema/defn ^:always-validate
+  describe-task :- TaskDetails
+  [task-data :- TaskData
+   env-name :- schema/Str
+   module-name :- schema/Str]
+  (let [?metadata (try (some-> task-data :metadata_file slurp json/decode)
```

Comment:
I don't have a strong preference either way, but since you only refer to metadata_file or files keys here, you could destructure that in the args list, especially since there's a schema explaining what you expect

Reply:
those are the only two keys in the map anyway. destructure or no destructure, I don't see much of a difference either way I guess.

Reply:
I've never used `?` to precede a varname before. Is that being used to indicate that it might be nil?

Reply:
@haus yup, that's the way I've seen it used. it's not *that* common, but it is a thing.